Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,234 @@
+                        float.Parse(userHistory[i + 1]) : float.NaN;
 
+                    if ((nextBalance - balance) < biggestLoss)
+                    {
+                        biggestLoser = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+                    if (biggestLoss == nextBalance - balance)
+                    {
+                        if (holderCount == 1)
+                        {
+                            biggestLoser += ($", {currentBalanceHolder}");
+                            holderCount++;
+                        }
+                        if (biggestLoss == nextBalance - balance &&
+                            biggestLoser != currentBalanceHolder &&
+                            holderCount > 1)
+                        {
+                            biggestLoser += ($" and {currentBalanceHolder}");
+                        }
+                    }
+                    biggestLoss = ((nextBalance - balance) < biggestLoss) ?
+                        (nextBalance - balance) : biggestLoss;
+                }
+            }
+            string stringBiggestLoss = Math.Abs(biggestLoss).ToString();
+            return ($"{biggestLoser} lost the most money. -¤{stringBiggestLoss}.");
+        }
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+            int holderCount = 0;
+            float highestLastBalance = float.MinValue;
+            string currentBalanceHolder;
+            string highestLastBalanceHolder = "";
+            foreach (string entry in peopleAndBalances)
+            {
+                string[] userHistory = entry.Split(',');
+
+                currentBalanceHolder = userHistory[0];
+
+                float currentBalanceHolderFinalBalance = float.Parse(userHistory[^1]);
+
+                if (highestLastBalance < currentBalanceHolderFinalBalance)
+                {
+                    highestLastBalanceHolder = currentBalanceHolder;
+                    holderCount = 1;
+                }
+
+                if (highestLastBalance == currentBalanceHolderFinalBalance &&
+                    highestLastBalanceHolder != currentBalanceHolder &&
+                    holderCount == 2)
+                {
+                    highestLastBalanceHolder += ($" and {currentBalanceHolder}");
+                }
+                if (highestLastBalance == currentBalanceHolderFinalBalance &&
+                   highestLastBalanceHolder != currentBalanceHolder &&
+                   holderCount == 1)
+                {
+                    highestLastBalanceHolder += ($", {currentBalanceHolder}");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use a StringBuilder instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)